File: packages/ts-client-sdk/src/sdk/models/operations/queryseatsbysubscription.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class QuerySeatsBySubscriptionPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscriptionId" })
  subscriptionId: string;
}

export class QuerySeatsBySubscriptionQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=userEmail" })
  userEmail?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=userId" })
  userId?: string;
}

export class QuerySeatsBySubscriptionRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: QuerySeatsBySubscriptionPathParams;

  @SpeakeasyMetadata()
  queryParams: QuerySeatsBySubscriptionQueryParams;
}

export class QuerySeatsBySubscriptionDefaultApplicationJSONIssues extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class QuerySeatsBySubscriptionDefaultApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code: string;

  @SpeakeasyMetadata({ data: "json, name=issues", elemType: QuerySeatsBySubscriptionDefaultApplicationJSONIssues })
  issues?: QuerySeatsBySubscriptionDefaultApplicationJSONIssues[];

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class QuerySeatsBySubscription200ApplicationJSONOccupant extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email: string;

  @SpeakeasyMetadata({ data: "json, name=seat_id" })
  seatId: string;

  @SpeakeasyMetadata({ data: "json, name=tenant_id" })
  tenantId: string;

  @SpeakeasyMetadata({ data: "json, name=user_id" })
  userId: string;

  @SpeakeasyMetadata({ data: "json, name=user_name" })
  userName: string;
}

export class QuerySeatsBySubscription200ApplicationJSONReservation extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email: string;

  @SpeakeasyMetadata({ data: "json, name=invite_url" })
  inviteUrl: string;

  @SpeakeasyMetadata({ data: "json, name=seat_id" })
  seatId: string;

  @SpeakeasyMetadata({ data: "json, name=tenant_id" })
  tenantId: string;

  @SpeakeasyMetadata({ data: "json, name=user_id" })
  userId: string;
}
export enum QuerySeatsBySubscription200ApplicationJSONSeatTypeEnum {
    Standard = "standard",
    Limited = "limited"
}
export enum QuerySeatsBySubscription200ApplicationJSONSeatingStrategyNameEnum {
    MonthlyActiveUser = "monthly_active_user",
    FirstComeFirstServed = "first_come_first_served"
}

export class QuerySeatsBySubscription200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=created_utc" })
  createdUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=expires_utc" })
  expiresUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=occupant" })
  occupant: QuerySeatsBySubscription200ApplicationJSONOccupant;

  @SpeakeasyMetadata({ data: "json, name=redeemed_utc" })
  redeemedUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=reservation" })
  reservation: QuerySeatsBySubscription200ApplicationJSONReservation;

  @SpeakeasyMetadata({ data: "json, name=seat_type" })
  seatType: QuerySeatsBySubscription200ApplicationJSONSeatTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=seating_strategy_name" })
  seatingStrategyName: QuerySeatsBySubscription200ApplicationJSONSeatingStrategyNameEnum;

  @SpeakeasyMetadata({ data: "json, name=subscription_id" })
  subscriptionId: string;
}

export class QuerySeatsBySubscriptionResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata({ elemType: QuerySeatsBySubscription200ApplicationJSON })
  querySeatsBySubscription200ApplicationJSONObjects?: QuerySeatsBySubscription200ApplicationJSON[];

  @SpeakeasyMetadata()
  querySeatsBySubscriptionDefaultApplicationJSONObject?: QuerySeatsBySubscriptionDefaultApplicationJSON;
}

File: packages/go-client-sdk/pkg/models/operations/queryseatsbysubscription.go
package operations

import (
	"time"
)

type QuerySeatsBySubscriptionPathParams struct {
	SubscriptionID string `pathParam:"style=simple,explode=false,name=subscriptionId"`
}

type QuerySeatsBySubscriptionQueryParams struct {
	UserEmail *string `queryParam:"style=form,explode=true,name=userEmail"`
	UserID    *string `queryParam:"style=form,explode=true,name=userId"`
}

type QuerySeatsBySubscriptionRequest struct {
	PathParams  QuerySeatsBySubscriptionPathParams
	QueryParams QuerySeatsBySubscriptionQueryParams
}

type QuerySeatsBySubscriptionDefaultApplicationJSONIssues struct {
	Message string `json:"message"`
}

type QuerySeatsBySubscriptionDefaultApplicationJSON struct {
	Code    string                                                 `json:"code"`
	Issues  []QuerySeatsBySubscriptionDefaultApplicationJSONIssues `json:"issues,omitempty"`
	Message string                                                 `json:"message"`
}

type QuerySeatsBySubscription200ApplicationJSONOccupant struct {
	Email    string `json:"email"`
	SeatID   string `json:"seat_id"`
	TenantID string `json:"tenant_id"`
	UserID   string `json:"user_id"`
	UserName string `json:"user_name"`
}

type QuerySeatsBySubscription200ApplicationJSONReservation struct {
	Email     string `json:"email"`
	InviteURL string `json:"invite_url"`
	SeatID    string `json:"seat_id"`
	TenantID  string `json:"tenant_id"`
	UserID    string `json:"user_id"`
}

type QuerySeatsBySubscription200ApplicationJSONSeatTypeEnum string

const (
	QuerySeatsBySubscription200ApplicationJSONSeatTypeEnumStandard QuerySeatsBySubscription200ApplicationJSONSeatTypeEnum = "standard"
	QuerySeatsBySubscription200ApplicationJSONSeatTypeEnumLimited  QuerySeatsBySubscription200ApplicationJSONSeatTypeEnum = "limited"
)

type QuerySeatsBySubscription200ApplicationJSONSeatingStrategyNameEnum string

const (
	QuerySeatsBySubscription200ApplicationJSONSeatingStrategyNameEnumMonthlyActiveUser    QuerySeatsBySubscription200ApplicationJSONSeatingStrategyNameEnum = "monthly_active_user"
	QuerySeatsBySubscription200ApplicationJSONSeatingStrategyNameEnumFirstComeFirstServed QuerySeatsBySubscription200ApplicationJSONSeatingStrategyNameEnum = "first_come_first_served"
)

type QuerySeatsBySubscription200ApplicationJSON struct {
	CreatedUtc          time.Time                                                         `json:"created_utc"`
	ExpiresUtc          time.Time                                                         `json:"expires_utc"`
	ID                  string                                                            `json:"id"`
	Occupant            QuerySeatsBySubscription200ApplicationJSONOccupant                `json:"occupant"`
	RedeemedUtc         time.Time                                                         `json:"redeemed_utc"`
	Reservation         QuerySeatsBySubscription200ApplicationJSONReservation             `json:"reservation"`
	SeatType            QuerySeatsBySubscription200ApplicationJSONSeatTypeEnum            `json:"seat_type"`
	SeatingStrategyName QuerySeatsBySubscription200ApplicationJSONSeatingStrategyNameEnum `json:"seating_strategy_name"`
	SubscriptionID      string                                                            `json:"subscription_id"`
}

type QuerySeatsBySubscriptionResponse struct {
	ContentType                                          string
	StatusCode                                           int64
	QuerySeatsBySubscription200ApplicationJSONObjects    []QuerySeatsBySubscription200ApplicationJSON
	QuerySeatsBySubscriptionDefaultApplicationJSONObject *QuerySeatsBySubscriptionDefaultApplicationJSON
}
